### PR TITLE
ProbeTerminationGracePeriod promote to GA

### PR DESCRIPTION
**Pull request by rphillips**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
https://github.com/kubernetes/kubernetes/pull/99375 introduced Probe-level terminationGracePeriodSeconds to provide a timeout for Liveness probes. The feature was initially targeted for GA in 1.25, but was missed.

The PR promotes ProbeTerminationGracePeriod to GA in 1.28 with a removal of the feature gate in 1.29.

#### Which issue(s) this PR fixes:
Fixes #

#### Does this PR introduce a user-facing change?
```release-note
Graduate the `ProbeTerminationGracePeriod` feature gate to GA
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs

```

### CODE REVIEW

**Comment by rphillips** [comment]
pull-kubernetes-node-e2e-containerd seems to be perma-failing on master branch.

**Comment by SergeyKanzhelev** [comment]
Let's make sure the KEP is approved for 1.27 to go to GA and passed the PRR.

Also, is there any tests that we need to promote to conformance with this change?

**Comment by sftim** [comment]
Suggestion for changelog entry:
```markdown
Graduate the `ProbeTerminationGracePeriod` feature gate to GA
```

**Comment by rphillips** [comment]
Updated the enhancement: https://github.com/kubernetes/enhancements/pull/3710

**Review by mrunalp — APPROVED** [review]

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -631,6 +631,7 @@ const (
 	// owner: @ehashman
 	// alpha: v1.21
 	// beta: v1.22
+	// ga: v1.27
```
Comment (SergeyKanzhelev) [inline pkg/features/kube_features.go:634]: is there any e2e tests that needs to be promoted to Conformance?

File: pkg/features/kube_features.go [hunk 2]
```diff
@@ -1001,7 +1002,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	PodSecurity: {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
 
-	ProbeTerminationGracePeriod: {Default: true, PreRelease: featuregate.Beta}, // Default to true in beta 1.25
+	ProbeTerminationGracePeriod: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.29
```
Comment (SergeyKanzhelev) [inline pkg/features/kube_features.go:1005]: when locking to true, remove all the mentions of the feature gate from code.

File: pkg/features/kube_features.go [hunk 3]
```diff
@@ -631,6 +631,7 @@ const (
 	// owner: @ehashman
```
Comment (SergeyKanzhelev) [inline pkg/features/kube_features.go:631]: feel free to put your name here for the history. Thank you for moving it forward

File: pkg/features/kube_features.go (hunk 1 above)
Comment (pacoxu) [reply pkg/features/kube_features.go:634]: Just work on that https://github.com/kubernetes/kubernetes/pull/116091 and find this PR.

**Comment by derekwaynecarr** [comment]
need to get the test case changes merged first, will come back to this pr.

**Comment by SergeyKanzhelev** [comment]
oh, we missed this for 1.27? I thought it is done. Can you also send the update for k/enhancements PR with the milestone update?

**Comment by rphillips** [comment]
@SergeyKanzhelev Done: https://github.com/kubernetes/enhancements/pull/3986

**Comment by pacoxu** [comment]
/lgtm
better merge with https://github.com/kubernetes/kubernetes/pull/116091

**Comment by dims** [comment]
@rphillips two things to follow up if needed 
- https://github.com/kubernetes/kubernetes/pull/114307#discussion_r1106294486
- https://github.com/kubernetes/kubernetes/pull/114307#discussion_r1106295157

Gonna approve this for now, please submit a follow up PRs if needed.

/approve
/lgtm

**Comment by rphillips** [comment]
@liggitt @smarterclayton could you approve for the top level directories?

**Comment by liggitt** [comment]
/approve

for API package updates

**Comment by gjkim42** [comment]
I guess this may be related to[ this regression](https://github.com/kubernetes/kubernetes/issues/119377).

File: pkg/api/pod/util.go [hunk 4]
```diff
@@ -495,7 +495,7 @@ func dropDisabledFields(
 		}
 	}
 
-	if !utilfeature.DefaultFeatureGate.Enabled(features.ProbeTerminationGracePeriod) && !probeGracePeriodInUse(oldPodSpec) {
+	if !probeGracePeriodInUse(oldPodSpec) {
```
Comment (gjkim42) [inline pkg/api/pod/util.go:498]: I think this is the problem.

We should never drop terminationGracePeriodSeconds of a probe.

I'll propose a fix for this.

Comment (gjkim42) [reply pkg/api/pod/util.go:498]: https://github.com/kubernetes/kubernetes/pull/119354

Comment (liggitt) [reply pkg/api/pod/util.go:498]: you are correct... this block should have gone away entirely

**Comment by sftim** [comment]
Changelog suggestion
```markdown
Graduated support for probe-level `terminationGracePeriodSeconds` to general availability.
```

We graduate _features_; feature gates change as part of this.

**Pull request merged**